src/hexo/themes/documentation/layout/partials/docsearch.hbs

```diff
@@ -1,10 +1,28 @@
 <script src="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js"></script>
 <script>
-    docsearch({
-        apiKey: '2ee58f908e6d720c4f72a9645ce0e857', // search-only API key, okay for frontend
-        indexName: 'sonarwhal',
-        inputSelector: '#search-input',
-        appId: '0R8E5SPAFP',
-        debug: false // set to true if need to debug css
-    });
+    (function () {
+        var search;
+        var previewSize = 500;
+        var transformInput = function () {
+            if (window.innerWidth > previewSize) {
+                // transform the input field to an algolia search input if there is enough space
+                search = docsearch({
+                    apiKey: '2ee58f908e6d720c4f72a9645ce0e857', // search-only API key, okay for frontend
+                    indexName: 'sonarwhal',
+                    inputSelector: '#search-input',
+                    appId: '0R8E5SPAFP',
```

Comment:
` apiKey`  and `appId` values are in 2 different places. They should be in the config file the same way we are using `theme.google_analytics` for google analytics.
Also there should be a way to read that configuration from the server side.